### DISCUSSION TITLE
Fullscreen canvas?

### DISCUSSION THREAD

**Discussion by tobi-wan-kenobi** [ArthurSonzogni/FTXUI#344] [Q&A] [answered]
Hello,

I am searching for a possibility to draw a full-screen canvas, but I am running into some issues.

I am instantiating an `ftxui::ScreenInteractive::Fullscreen()` and later, in the renderer, use `screen.dimx()` and `screen.dimy()` to create the canvas.

Once the first event (keypress,...) is registered, the canvas resizes (from 0/0), but it's less than expected - on the horizontal, I'd estimate it's half of the full screen, on the vertical, it's hard to say - maybe a fourth?

I also tried playing around with `ftxui::flex` (thinking that maybe this helps), but no dice. The canvas is inside an `ftxui::vbox`, in case that makes a difference.

Any help would be much appreciated!

**Comment by ArthurSonzogni** [comment, marked as answer]
When using braille characters, we can display:
- 1x1 cells per cells
- 1x2 blocks by cells
- 2x4 brailles dot per cells.
This explains why you see a difference of coordinate/size differences. The canvas uses a coordinate system scaled by 2x4.

What about using:
```cpp

auto component = Renderer([] {
  auto my_image = canvas([] (Canvas& c) {
    c.DrawPointLine(0, 0, canvas.width(), canvas.height());
    c.DrawText(0, 0, "width = " + std::to_string(canvas.width()));
    c.DrawText(0, 2, "height = " + std::to_string(canvas.height()));
  });
  return my_image | flex;
}

auto screen = ScreenInteractive::Fullscreen();
```

This create a component, that will display a canvas "flexible", taking the whole size of the screen.

**Reply by tobi-wan-kenobi** [reply]
Ah, I see.

Many thanks for the detailed answer, I like the `flex` approach a lot!

Very, very cool project, by the way!

**Reply by ArthurSonzogni** [reply]
Thank you!
